genome_designer/genome_finish/insertion_placement_read_trkg.py

```diff
@@ -473,12 +474,7 @@ def write_read_query_alignments_to_fastq(reads, fastq_path,
 def simple_align_with_bwa_mem(reads_fq, reference_fasta, output_bam_path):
 
     # Ensure reference fasta is indexed
-    subprocess.call(' '.join([
-            '%s/bwa/bwa' % settings.TOOLS_DIR,
-            'index',
-            reference_fasta
-            ]),
-            shell=True, executable=settings.BASH_PATH)
+    ensure_bwa_index(reference_fasta)
```

Comment:
This function might actually try to index and cause the threading issue we were debugging. Instead we should just have a hard assert here that checks for the existence of the index. Maybe make a new function in read_alignment_util.py called something like `has_bwa_index()` that returns a boolean which is the result of calling ` os.path.exists(ref_genome_fasta + '.bwt')`. And here, just `assert` on that result. If that fails, we have another bug somewhere above but we'll catch that immediately.